en/docs/install-and-setup/upgrading-wso2-api-manager/260-to-410/upgrading-from-260-to-410.md

```diff
@@ -0,0 +1,1304 @@
+
+    !!! note
+        Log4j2 has hot deployment support, and **Managing Logs** section has been removed from the Management Console. You can now use the log4j2.properties file to modify logging configurations without restarting the server.
+
+5.  Copy the keystores (i.e., `client-truststore.jks`, `wso2cabon.jks` and any other custom JKS) used in the previous version and replace the existing keystores in the `<API-M_4.1.0_HOME>/repository/resources/security` directory.
+
+    !!! Attention
+        In API Manager 4.1.0, it is required to use a certificate with the RSA key size greater than 2048. If you have used a certificate that has a weak RSA key (key size less than 2048) in previous version, you need to add the following configuration to `<API-M_4.1.0_HOME>/repository/conf/deployment.toml` file to configure internal and primary keystores. You should point the internal keystore to the keystore copied from API Manager 2.6.0 and the primary keystore can be pointed to a keystore with a certificate, which has a strong RSA key.
+    
+        ``` java
+            [keystore.tls]
+            file_name = "internal.jks"
+            type = "JKS"
+            password = "wso2carbon"
+            alias = "wso2carbon"
+            key_password = "wso2carbon"
+                    
+            [keystore.primary]
+            file_name = "primary.jks"
+            type = "JKS"
+            password = "wso2carbon"
+            alias = "wso2carbon"
+            key_password = "wso2carbon"
+    
+            [keystore.internal]
+            file_name = "internal.jks"
+            type = "JKS"
+            password = "wso2carbon"
+            alias = "wso2carbon"
+            key_password = "wso2carbon"
+        ```
+
+    !!! note "If you have enabled Secure Vault"
+        If you have enabled secure vault in the previous API-M version, you need to add the property values again according to the new config modal and run the script as below. Refer [Encrypting Passwords in Configuration files]({{base_path}}/install-and-setup/setup/security/logins-and-passwords/working-with-encrypted-passwords) for more details.
+    
+        ```tab="Linux"
+        ./ciphertool.sh -Dconfigure
+        ```
+
+        ```tab="Windows"
+        ./ciphertool.bat -Dconfigure
+        ```
+
+    - In order to work with the [API Security Audit Feature]({{base_path}}/design/api-security/configuring-api-security-audit/) you need to have the public certificate of the [42crunch](https://42crunch.com/) in the client-truststore. Follow the guidelines given in [Importing Certificates to the Truststore]({{base_path}}/install-and-setup/setup/security/configuring-keystores/keystore-basics/creating-new-keystores/#step-3-importing-certificates-to-the-truststore).
+
+6. If you already have used secondary user stores in API-M 3.2.0, you have to copy the user store file created inside the old API-M version to the new APIM version.
```

Comment:
API-M 2.6.0

Reply:
fixed in https://github.com/wso2/docs-apim/pull/5985/commits/7ebc3082303ca860fea559dbc70e26b6ed238fad#diff-b598cabec9a35d16c1c419398bd373cca84f31896a22a34bc24448b9606d99b0R474